src/Microsoft.Framework.Configuration.Abstractions/IConfigurationBuilder.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Framework.Configuration
 {
     public interface IConfigurationBuilder
     {
-        string BasePath { get; }
+        Dictionary<string, object> Properties { get; }
```

Comment:
I'm a fan of just `Properties` - that's a pretty common name used in lots of places for "a bag of stuff."